src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -56,11 +58,30 @@
       ;; Respond to all CSR validation failures with a 400
       (middleware-utils/plain-response 400 msg))))
 
-(defn handle-get-certificate-revocation-list
-  [{:keys [cacrl]}]
-  (-> (ca/get-certificate-revocation-list cacrl)
-      (rr/response)
-      (rr/content-type "text/plain")))
+      ;;A recipient MUST ignore If-Modified-Since if the request contains an
+      ;; If-None-Match header field;
+  (defn handle-get-certificate-revocation-list
+    "Always return the crl if no 'if-modified-since' header is provided or if that header is not in correct http-date format.
+    If the header is present and correctly formatted, only return the crl if the master cacrl is newer than the agent crl."
+    [request {:keys [cacrl]}]
+    (let [agent-crl-last-modified-val (rr/get-header request "If-Modified-Since")
+          agent-crl-last-modified-date-time (if agent-crl-last-modified-val
+                                              (try
+                                                (time-format/parse
+                                                  (time-format/formatters :rfc822)
+                                                  (string/replace agent-crl-last-modified-val #"GMT" "+0000"))
+                                                (catch Exception e
+                                                  :unknown))
+                                              :unknown)
```

Comment:
I think we should put this processing in its own function, there's a lot going on here with the `if` and the `try`.

Also, is it possible to be more specific about what exception we expect, i.e. is there a more specific `Exception` type that is being thrown by the `string/replace`?

Reply:
Will move to own function.  It will be an exception raised by (time/parse my-custom-formatter).  Lemme see if it makes sense to be more specific there.

Reply:
Yeah, I think so. That would make it clearer what went wrong that we are squashing.

Reply:
Ok.  separated it out into its own function and catching that specific error now.